data/creaturescripts/scripts/logout.lua

```diff
@@ -3,5 +3,19 @@ function onLogout(player)
 	if nextUseStaminaTime[playerId] ~= nil then
 		nextUseStaminaTime[playerId] = nil
 	end
+
+	local stats = player:inBossFight()
+	if stats then
+		-- Player logged out (or died) in the middle of a boss fight, store his damageOut and stamina
+		local boss = Monster(stats.bossId)		
+		if boss then
+			local dmgOut = boss:getDamageMap()[playerId]
```

Comment:
damageTaken I suppose, just avoid writing names like p insted of player, it's fine if it's your own server since you coded it - you know what everything is.
But just try to convert 0.x scripts to 1.x, where all variables are named b_2 etc
As long as the name describes what is stored in the variable it's fine.